src/main/java/io/supertokens/oauth/exceptions/OAuthException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ *    Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ *    This software is licensed under the Apache License, Version 2.0 (the
+ *    "License") as published by the Apache Software Foundation.
+ *
+ *    You may not use this file except in compliance with the License. You may
+ *    obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.oauth.exceptions;
+
+import java.io.Serial;
+
+public class OAuthException extends Exception{
```

Comment:
I dont think you need this class. All OAuth related exceptions can directly inherit from Exception

Reply:
This is used to make sure all the subclasses have `error` and `errorDescription` fields. Hydra generally returns errors as some 400 status code and a json body containing `error` and `error_description` fields. I created a method to automatically map our HttpResponseExeption to the subclasses of this OAuthException, but for that I needed to make sure that all the hydra related exceptions have `error` and `errorDescription` fields.